coreneuron/nrniv/nrn_setup.cpp

```diff
@@ -49,6 +49,85 @@ THE POSSIBILITY OF SUCH DAMAGE.
+
+void (*nrn2core_get_partrans_setup_info_)(int tid,
+                                          int& ntar,
+                                          int& nsrc,
+                                          int& type,
+                                          int& ix_vpre,
+                                          int*& sid_target,
+                                          int*& sid_src,
+                                          int*& v_indices);
+
+int (*nrn2core_get_dat1_)(int tid,
+                          int& n_presyn,
+                          int& n_netcon,
+                          int*& output_gid,
+                          int*& netcon_srcgid);
+
+int (*nrn2core_get_dat2_1_)(int tid,
+                            int& ngid,
+                            int& n_real_gid,
+                            int& nnode,
+                            int& ndiam,
+                            int& nmech,
+                            int*& tml_index,
+                            int*& ml_nodecount,
+                            int& nidata,
+                            int& nvdata,
+                            int& nweight);
+
+int (*nrn2core_get_dat2_2_)(int tid,
+                            int*& v_parent_index,
+                            double*& a,
+                            double*& b,
+                            double*& area,
+                            double*& v,
+                            double*& diamvec);
+
+int (*nrn2core_get_dat2_mech_)(int tid,
+                               size_t i,
+                               int dsz_inst,
+                               int*& nodeindices,
+                               double*& data,
+                               int*& pdata);
+
+int (*nrn2core_get_dat2_3_)(int tid,
+                            int nweight,
+                            int*& output_vindex,
+                            double*& output_threshold,
+                            int*& netcon_pnttype,
+                            int*& netcon_pntindex,
+                            double*& weights,
+                            double*& delays);
+
+int (*nrn2core_get_dat2_corepointer_)(int tid, int& n);
+
+int (*nrn2core_get_dat2_corepointer_mech_)(int tid,
+                                           int type,
+                                           int& icnt,
+                                           int& dcnt,
+                                           int*& iarray,
+                                           double*& darray);
+
+int (*nrn2core_get_dat2_vecplay_)(int tid, int& n);
+
+int (*nrn2core_get_dat2_vecplay_inst_)(int tid,
+                                       int i,
+                                       int& vptype,
+                                       int& mtype,
+                                       int& ix,
+                                       int& sz,
+                                       double*& yvec,
+                                       double*& tvec);
+
```

Comment:
if these are declared in nrn2core_direct.h, why we need to declare here again?

Reply:
they are all extern. the file is included in multiple places.